AlexMarrinan/RavenGuard
# [RGT-78] Basic XP and level ups

Testing:
- Defeat an enemy
- You will gain 100xp, causing you to level up
- A level up screen should appear
- Choose 2 stats to increase your unit in, the value added is random from 1-4

TODO (probably in future PR):
- Add unit stat proficiencies to determine the random values to choose from for stat upgrades
- Add animation for xp bar increasing when gaining xp
- Add animation when the xp bar reaches a new level
- Improve UI with animations and correct art assets
